hathor/transaction/storage/traversal.py

```diff
@@ -160,13 +177,39 @@ def _pop_visit(self) -> 'BaseTransaction':
         return tx
 
 
+class BFSOrderWalk(GenericWalk):
+    """ A help to walk in the DAG using a BFS.
+    """
+    _to_visit: deque['BaseTransaction']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._to_visit = deque()
+
+    def _is_empty(self) -> bool:
+        return not self._to_visit
+
+    def _push_visit(self, tx: 'BaseTransaction') -> None:
+        self._to_visit.append(tx)
+
+    def _pop_visit(self) -> 'BaseTransaction':
+        return self._to_visit.popleft()
+
+
 class DFSWalk(GenericWalk):
     """ A help to walk in the DAG using a DFS.
     """
-    to_visit: list['BaseTransaction']
+    _to_visit: list['BaseTransaction']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._to_visit = []
+
+    def _is_empty(self) -> bool:
+        return not self._to_visit
```

Comment:
It looks like this method is the same in all subclasses, what about moving it to the superclass?

Reply:
The superclass doesn't know about `self._to_visit`, which has different types depending on the subclass, and in theory the subclass could use something else entirely. I'd say it's mostly a coincidence that this works out exactly the same on the three of them.

Reply:
Ok!